domains/runtime/system/src/runtime.rs

```diff
@@ -620,6 +621,22 @@ impl_runtime_apis! {
         }
     }
 
+    impl sp_domains::transaction::PreValidationObjectApi<Block, domain_runtime_primitives::Hash> for Runtime {
+        fn extract_pre_validation_object(
+            extrinsic: <Block as BlockT>::Extrinsic,
+        ) -> PreValidationObject<Block, domain_runtime_primitives::Hash> {
+                match extrinsic.function {
+                    RuntimeCall::DomainRegistry(pallet_domain_registry::Call::submit_fraud_proof { fraud_proof }) => {
+                        PreValidationObject::FraudProof(fraud_proof)
+                    }
+                    RuntimeCall::DomainRegistry(pallet_domain_registry::Call::submit_core_bundle {
+                        signed_opaque_bundle,
+                    }) => PreValidationObject::Receipts(signed_opaque_bundle.bundle.receipts),
```

Comment:
This call will never appear in the transaction pool as it's only crafted and injected on fly when building the system domain block. We may rename `submit_core_bundle` to something more explicit to imply this point.

```suggestion
```

Reply:
Right, the term `submit` could be ambiguous. I will leave a TODO from now.